Core/src/org/sleuthkit/autopsy/modules/filetypeid/UserDefinedFileTypesManager.java

```diff
@@ -231,17 +231,47 @@ private void loadUserDefinedFileTypes() throws UserDefinedFileTypesException {
      * @param fileType The file type to add.
      */
     private void addUserDefinedFileType(FileType fileType) {
-        userDefinedFileTypes.put(fileType.getMimeType(), fileType);
-        fileTypes.put(fileType.getMimeType(), fileType);
+        addFileTypeToMap(userDefinedFileTypes, fileType);
+        addFileTypeToMap(fileTypes, fileType);
+    }
+
+    /**
+     * Adds given FileType to given map:
+     * if the mimetype exists, add fileType to that mimetype's list.
+     * otherwise, create a new mimetype with the new fileType.
+     * @param map The map to be modified.
+     * @param fileType The added FileType
+     */
+    void addFileTypeToMap(Map<String, List<FileType>> map, FileType fileType) {
+        String mimeType = fileType.getMimeType();
```

Comment:
guava provides a great MultiMap (see https://code.google.com/p/guava-libraries/wiki/NewCollectionTypesExplained) that is quite usefull in this kind of situation.